### PR TITLE
Implement tasks homework_04

### PR DESCRIPTION
**Pull request by AnastasiyaGavrilik**

### CODE REVIEW

**Comment by vladhilko** [comment]
Please, run `rubocop-rspec` 😔

**Comment by AnastasiyaGavrilik** [comment]
@vladhilko  Done🙂

File: HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+RSpec.describe Custom do
+  describe 'my_each' do
```
Comment (vladhilko) [inline HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb:2]: http://www.betterspecs.org/#describe
```suggestion
  describe '#my_each' do
```

File: HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    let(:arr) { [1, 2, 3] }
+
+    before { Array.include(described_class) }
+
+    context 'when empty block given' do
+      subject { arr.my_each }
+
+      it 'returns self' do
+        expect(subject).to eq arr
+      end
```
Comment (vladhilko) [inline HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb:12]: We can use one line syntax 😉
```suggestion
      it { is_expected.to eq arr }
```

File: HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    end
+
+    context 'when block given' do
+      subject { arr.my_each { |el| check_arr << (el * 2) } }
+
+      let(:check_arr) { [] }
+      let(:expected_arr) { [2, 4, 6] }
+
+      it 'block works and returns self' do
```
Comment (vladhilko) [inline HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb:21]:
```suggestion
      it 'works and returns self' do
```

File: HW04/Anastasiya_Gavrilik/spec/custom/my_map_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+RSpec.describe Custom do
+  describe 'my_map' do
```
Comment (vladhilko) [inline HW04/Anastasiya_Gavrilik/spec/custom/my_map_spec.rb:2]: The same comments for this test 🙃

File: HW04/Anastasiya_Gavrilik/spec/custom/my_select_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+RSpec.describe Custom do
+  describe 'my_select' do
```
Comment (vladhilko) [inline HW04/Anastasiya_Gavrilik/spec/custom/my_select_spec.rb:2]: The same comments for this test 🙃

File: HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb (hunk 1 above)
Comment (AnastasiyaGavrilik) [reply HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb]: Done, thanks

File: HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb (hunk 2 above)
Comment (AnastasiyaGavrilik) [reply HW04/Anastasiya_Gavrilik/spec/custom/my_each_spec.rb]: thanks for your  advice

**Pull request merged**